src/components/code-line.component.js

```diff
@@ -63,44 +63,79 @@ const styles = StyleSheet.create({
   },
 });
 
-export const CodeLine = ({ newChunk, change }: Props) =>
-  <View style={styles.container}>
-    <View style={styles.wrapper}>
-      <View
-        style={[
-          styles.lineNumbers,
-          newChunk && styles.newChunkLineNumbers,
-          change.type === 'add' && styles.addLineNumbers,
-          change.type === 'del' && styles.delLineNumbers,
-        ]}
-      >
-        <Text style={styles.codeLineNumber}>
-          {change.type === 'del'
-            ? change.ln
-            : change.type === 'normal'
-              ? change.ln1
-              : change.type === 'add' ? '' : '...'}
-        </Text>
-        <Text style={styles.codeLineNumber}>
-          {change.type === 'add'
-            ? change.ln
-            : change.type === 'normal'
-              ? change.ln2
-              : change.type === 'del' ? '' : '...'}
-        </Text>
-      </View>
+export class CodeLine extends Component {
+  props: {
+    newChunk: boolean,
+    change: Object,
+    filename: string,
+  };
+
+  isKnownType(language) {
+    return language && getLanguage(language) && !this.props.newChunk;
+  }
+
+  render() {
+    const { newChunk, change, filename } = this.props;
+    const language = (filename || []).split('.').pop();
+    // SyntaxHighlighter doesn't allow Stylesheet class style
+    const customStyle = {
+      backgroundColor: 'transparent',
+      padding: 0,
+    };
+
+    return (
+      <View style={styles.container}>
+        <View style={styles.wrapper}>
+          <View
+            style={[
+              styles.lineNumbers,
+              newChunk && styles.newChunkLineNumbers,
+              change.type === 'add' && styles.addLineNumbers,
+              change.type === 'del' && styles.delLineNumbers,
+            ]}
+          >
+            <Text style={styles.codeLineNumber}>
+              {change.type === 'del'
+                ? change.ln
+                : change.type === 'normal'
+                  ? change.ln1
+                  : change.type === 'add' ? '' : '...'}
+            </Text>
+            <Text style={styles.codeLineNumber}>
+              {change.type === 'add'
+                ? change.ln
+                : change.type === 'normal'
+                  ? change.ln2
+                  : change.type === 'del' ? '' : '...'}
+            </Text>
+          </View>
+
+          <View
+            style={[
+              styles.codeLineContainer,
+              newChunk && styles.newChunkLineContainer,
+              change.type === 'add' && styles.addLine,
+              change.type === 'del' && styles.delLine,
+            ]}
+          >
+            {(newChunk || !this.isKnownType(language)) &&
```

Comment:
`isKnownType(language)` function checks if it isn't a `newChunk` (line 74). So if `newChunk === true`, `this.isKnownType(language)` can't be `true` !